src/core/activity/activityChart/activityChartPanel.ts

```diff
@@ -0,0 +1,309 @@
+import * as math from 'mathjs';
+
+import { ActivityChartPanelModel } from './activityChartPanelModel';
+import { ActivityChartGraph } from './activityChartGraph';
+
+import { AnalogSignalPlotModel } from './activityChartPanelModels/analogSignalPlotModel';
+import { AnalogSignalHistogramModel } from './activityChartPanelModels/analogSignalHistogramModel';
+import { SpikeTimesRasterPlotModel } from './activityChartPanelModels/spikeTimesRasterPlotModel';
+import { SpikeTimesHistogramModel } from './activityChartPanelModels/spikeTimesHistogramModel';
+import { SpikeSendersHistogramModel } from './activityChartPanelModels/spikeSendersHistogramModel';
+import { InterSpikeIntervalHistogramModel } from './activityChartPanelModels/interSpikeIntervalHistogramModel';
+import { CVISIHistogramModel } from './activityChartPanelModels/CVISIHistogramModel';
+
+export class ActivityChartPanel {
+  // private static readonly _name = 'ActivityGraphPanel';
+  // private _activities: Activity[] = [];
+  private _graph: ActivityChartGraph; // parent
+  private _layout: any = {
+    xaxis: {
+      showgrid: true,
```

Comment:
During the update all layouts from various panels (`activityChartPanel.ts`) will be collected and stored in `activityChartGraph.ts`, then it will forward to Plotly component in `ActivityGraph.vue`.